### ISSUE TITLE
Flatten page structure - Error for content mixing template and components

### ISSUE THREAD

**Issue by GlennRicaud** [enonic/xp#6820]
There seems to be a problem when displaying content with both template and components
* Use app-superhero in branch issue-18
** The blog posts are correctly displayed but the main page /superhero is not correct. Its components are not displayed.

Investigate and fix

**Comment by sigdestad** [comment]
Are you sure there are no changes required to the templates? Remember superhero is really old...

**Comment by GlennRicaud** [comment]
Yes maybe it is because the data in superhero is obsolete. That is what has to be found here: if superhero was incorrect (and needs to be fixed) or if there is a case that is not handled here (and then to be fixed).

To test both existing and new behaviour, work with:
1) xp in branch ``master``and app-superhero-blog in branch ``master``
2) xp in branch ``epic-6649``and app-superhero-blog in branch ``issue-18``

**Comment by GlennRicaud** [comment]
@ashklianko :
I reproduced your problem of import of superhero without the components. You are right superhero is not imported correctly:

This is due to the call to contentService.update in com.enonic.xp.demo.initializer.DemoInitializer in app-superhero. This call removes the other components.

But even when imported correctly the page is not correctly displayed.

So 2 bugs here actually:
- Page with template AND components
- ContentService.update

**Comment by ashklianko** [comment]
Hi Glenn, found where it comes from: PageDataSerializer.getPageRegions() needs a descriptor to fetch regions (I totally forgot about it), and root node in superhero is having only a template; Thus we end up with regions not being imported properly (or properly and then regions wiped out with update).
I've created PR for superhero: https://github.com/enonic/app-superhero-blog/pull/19
Can we say that this is expected behavior for new page structure?
